disassemblers/ofrak_angr/ofrak_angr/components/angr_decompilation_analyzer.py

```diff
@@ -0,0 +1,67 @@
+from dataclasses import dataclass
+import angr
+from angr.analyses.decompiler import Decompiler
+from ofrak.component.analyzer import Analyzer
+from ofrak.component.identifier import Identifier
+from ofrak.resource_view import ResourceView
+
+from ofrak.resource import Resource
+from ofrak.core.complex_block import ComplexBlock
+from ofrak.service.resource_service_i import ResourceFilter
+from ofrak_angr.model import AngrAnalysis, AngrAnalysisResource
+
+
+@dataclass
+class AngrDecompilationAnalysis(ResourceView):
+    decompilation: str
+
+
+class AngrDecompilationAnalysisIdentifier(Identifier):
+    id = b"AngrDecompilationAnalysisIdentifier"
+    targets = (ComplexBlock,)
+
+    async def identify(self, resource: Resource, config=None):
+        resource.add_tag(AngrDecompilationAnalysis)
+
+
+class AngrDecompilatonAnalyzer(Analyzer[None, AngrDecompilationAnalysis]):
+    id = b"AngrDecompilationAnalyzer"
+    targets = (ComplexBlock,)
+    outputs = (AngrDecompilationAnalysis,)
+
+    async def analyze(self, resource: Resource, config: None) -> AngrDecompilationAnalysis:
+        # Run / fetch angr analyzer
+        try:
+            root_resource = await resource.get_only_ancestor(
+                ResourceFilter(tags=[AngrAnalysisResource], include_self=True)
+            )
+            complex_block = await resource.view_as(ComplexBlock)
+            angr_analysis = await root_resource.analyze(AngrAnalysis)
+
+            cfg = angr_analysis.project.analyses[angr.analyses.CFGFast].prep()(
+                data_references=True, normalize=True
+            )
+
+            function_s = [
+                func
+                for addr, func in angr_analysis.project.kb.functions.items()
+                if func.addr == complex_block.virtual_address
+            ]
+            if len(function_s) != 1:
+                # Check for thumb
+                function_s = [
+                    func
+                    for addr, func in angr_analysis.project.kb.functions.items()
+                    if func.addr == complex_block.virtual_address + 1
+                ]
```

Comment:
I don't understand why you only check for THUMB if the length of the list is not 1? Is `len(function_s)` only supposed to be either 0 or 1?

Reply:
Ok, this really should be if len(function_s) == 0, meaning that if there is no function at the desired address, try again with address+1 for thumb intstructions.